### PR TITLE
chore: Remove unused StringView struct

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion-comet/issues/1141

## Rationale for this change

I was looking into the report of unsafe behavior of StringView and was not familiar with this struct and do not see where it is used, so perhaps this is legacy code that is no longer needed? I am assuming that CI will tell me if this is actually still needed.

## What changes are included in this PR?

Remove `StringView` struct.

### CODE REVIEW

**Review by viirya — APPROVED** [review]

File: native/core/src/data_type.rs [hunk 1]
```diff
@@ -1,241 +0,0 @@
...
-make_type_trait!(ShortType, i16, TypeKind::Short);
-make_type_trait!(IntegerType, i32, TypeKind::Integer);
-make_type_trait!(LongType, i64, TypeKind::Long);
-make_type_trait!(FloatType, f32, TypeKind::Float);
-make_type_trait!(DoubleType, f64, TypeKind::Double);
-make_type_trait!(DecimalType, i128, TypeKind::Decimal);
-make_type_trait!(StringType, StringView, TypeKind::String);
-make_type_trait!(BinaryType, StringView, TypeKind::Binary);
-make_type_trait!(TimestampType, i64, TypeKind::Timestamp);
-make_type_trait!(DateType, i32, TypeKind::Date);
```
Comment (viirya) [inline native/core/src/data_type.rs:241]: I forgot that why @sunchao added these traits.

Comment (sunchao) [reply native/core/src/data_type.rs:241]: oops this was for an experiment to introduce Comet specific array vectors, in order to make vectorized hash table in aggregation work ..

Comment (viirya) [reply native/core/src/data_type.rs:241]: Okay, so I guess that we can remove them now?

Comment (sunchao) [reply native/core/src/data_type.rs:241]: yea, they are not used anymore

Comment (viirya) [reply native/core/src/data_type.rs:241]: Thanks for confirming it.

**Pull request merged**